content/what-is-jsx.md

```diff
@@ -0,0 +1,250 @@
+</MyComponent>
+```
+
+will become
+
+```javascript
+createElement(MyComponent, { title: title },
+    createElement("p", null, description));
+```
+
+Thus we use JavaScript expressions for conditionals:
+
+```jsx
+<MyComponent title={title || "unknown"}>
+    {description && (
+        <p>{description}</p>
+    )}
+</MyComponent>
+```
+
+Similarly, loops are expressions as well:
+
+```jsx
+<ul>
+    {items.map(item => (
+        <li>{item}</li>
+    ))}
+</ul>
+```
+
+### Special Cases for Boolean Attributes
+
+JSX supports a shorthand for boolean parameters, thus the following two
+statements are identical:
+
+```jsx
+<button disabled>
+```
+
+```jsx
+<button disabled={true}>
+```
+
+### Trying It Out
+
+You can try out how JSX translation works in the
+[Babel online compiler](https://babeljs.io/repl#?babili=false&browsers=&build=&builtIns=false&spec=false&loose=false&code_lz=DwWQngwg9gtgDlAdgU0QFwARoJZoDbIC8ARABLJ55QYDqUATngCbEB8AUAJDBytX3IYGbHADOAVxjAA9L3Yzw0eElRpWQA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=es2015%2Creact%2Cstage-2&prettier=false&targets=&version=7.7.3&externalPlugins=).
+It translates it using the `createElement` function from the React library,
+but we can also tell Babel to use a different `createElement` function from a
+library like complate.
+
+
+complate's `createElement` implementation
+-----------------------------------------
+
+Now we want to look at the specifics of complate's own JSX implementation.
+
+We saw before that
+
+```jsx
+<MyComponent title="Hello World">
+    <p>lorem ipsum</p>
+<MyComponent>
+```
+
+becomes
+
+```javascript
+createElement(MyComponent, { title: title },
+    createElement("p", null, description));
+```
+
+In order for this to work, the complate `createElement` function needs
+to be available in the JavaScript scope.
+
+`createElement` in complate expects a user defined component (called a macro)
+to be a function with the signature `(params, ...children)` and to return a JSX
+expression. An example definition for `MyComponent` could be:
+
+```jsx
+function MyComponent({ title }, ...children) {
+    return <article>
+        <h2>{title}</h2>
+        {children}
+    </article>;
+}
+```
+
+This is a complate specific implementation detail: in other JSX implementations
+a user defined component could be a class, but the JSX spec itself does not make
+any statement about what a user defined component is, other than that it should
+be in the scope of the application.
+
+### Dealing with Boolean Parameters
+
+complate ensures that boolean parameters in macros are correctly translated to
+HTML boolean attributes. This means that the JSX `<div hidden={expression}>`
+will be correctly translated to `<div hidden>` or `<div>` based on the
+result of the JavaScript `expression` that we evaluate. complate considers
+"blank" values (`false`, `undefined`, `null`) to be falsey.
```

Comment:
This seems potentially misleading: complate does _not_ know which HTML attributes are boolean (e.g. `disabled`, `hidden`) and which aren't (e.g. `aria-disabled`, `aria-hidden`). Instead, it merely turns a `true` value into a boolean attribute (`<… disabled>`) and omits such attributes for falsy values.

How about something like this:

> complate turns boolean parameter values into corresponding HTML attributes. Thus `<div hidden={true}>` becomes `<div hidden>` whereas `<div hidden={false}>` becomes simply `<div>` - yet `<div aria-hidden="true">` remains just that. Note that for convenience, "blank" values (`false`, `undefined`, `null`) are all considered falsey in this context.